### ISSUE TITLE
[Feature] Required baseOptions/variables in generated hooks of typescript-react-apollo when a query or mutation is created with required parameters

### ISSUE THREAD

**Issue by anderha** [dotansimha/graphql-code-generator#4964]
Hey,

I am using graphql-code-generator for typescript and react with version 2.0.7 of graphql-codegen/typescript-react-apollo and apollo/client version 3.2.2.

As in the discussion "[How to configure baseOptions as required?](https://github.com/dotansimha/graphql-code-generator/discussions/4887#discussion-24197)" stated I would really appreciate if  baseOptions and variables could be marked as required by the code generator when I create a query or mutation with required parameters. See the example below or in the discussion for clarification. I would really like to see this feature for queries and mutations.

My types:
```graphql
type Query {
    todo(id: Long!): Todo!
}

type Todo {
    id: Long
    description: String
    done: Boolean
}
```

My query: 
```graphql
query Todo($id: Long!) {
    todo(id: $id) {
        description,
        done
    }
}
```

Generated function:
```ts
export function useTodoQuery(baseOptions?: Apollo.QueryHookOptions<TodoQuery, TodoVariables>) {
    return Apollo.useQuery<TodoQuery, TodoVariables>(TodoDocument, baseOptions);
}
```

Call the hook:
```ts
// query with required parameter:
const { loading, error, data } = useTodoQuery({ variables: {id: 1}})

// query without required parameter:
const { loading, error, data } = useTodoQuery() // no error, because baseOptions are optional -> but I do want an error if i have to hand parameters over
```

**Comment by dotansimha** [comment]
Thank you for reporting this @anderha . 
I added support for this in this PR:  https://github.com/dotansimha/graphql-code-generator/pull/5037

Can you please try it? the alpha version is `@graphql-codegen/typescript-react-apollo@2.1.0-alpha-bd361352.0`.

**Comment by dotansimha** [comment]
Available in `@graphql-codegen/typescript-react-apollo@2.1.0`

**Comment by gynekolog** [comment]
Is there an option to disable this feature?

Example:
```ts
const [login] = useLoginMutation();

const handleSubmit = useCallback(async () => {
    try {
      const token = await getToken();
      await login({
        variables: { input: { authToken: token.accessToken } }
      });
    } catch (error) {
      console.log(error);
    }
  }, [login]);
```

`const [login] = useLoginMutation()` throw TS error now.
It can be solved by `const [login] = useLoginMutation({})` but I don't want make this "fix" across the whole project.
Is solution of this issue at @graphql-codegen/typescript-react-apollo@2.1.0 correct when error can be resolved by empty object?

**Comment by dotansimha** [comment]
@gynekolog thank you for reporting this. We are tracking the improvement of this feature here: https://github.com/dotansimha/graphql-code-generator/issues/5045

**Comment by anderha** [comment]
With the fixed mutation parameters in [#5045 ](https://github.com/dotansimha/graphql-code-generator/issues/5045#issue-736843294) it works like charm. Thanks a lot for this improvement @dotansimha
